1/contrib/s2i/assemble

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+JENKINS_DIR=/opt/openshift
+JENKINS_INSTALL_DIR=$(mktemp -d --suffix=jenkins)
+
+echo "---> Copying repository files ..."
+cp -Rf /tmp/src/. ${JENKINS_INSTALL_DIR}
+
+# Install jenkins plugins using plugin installer if user has "plugins.txt" file
+# in repository
+if [ -f ${JENKINS_INSTALL_DIR}/plugins.txt ]; then
+  echo "---> Installing Jenkins $(cat ${JENKINS_INSTALL_DIR}/plugins.txt|wc -l) plugins using ${JENKINS_DIR}/plugins.txt ..."
+  /usr/local/bin/plugins.sh ${JENKINS_INSTALL_DIR}/plugins.txt
+  chmod -R go+rw ${JENKINS_DIR}/plugins
+fi
+
+if [ -d ${JENKINS_INSTALL_DIR}/plugins ]; then
+  echo "---> Installing $(ls -l ${JENKINS_INSTALL_DIR}/plugins | grep ^- | wc -l) Jenkins plugins from plugins/ directory ..."
+  cp -R ${JENKINS_INSTALL_DIR}/plugins/* ${JENKINS_DIR}/plugins/
+  chmod -R go+rw ${JENKINS_DIR}/plugins
+fi
+
+if [ -d ${JENKINS_INSTALL_DIR}/configuration ]; then
+  if [ -d ${JENKINS_INSTALL_DIR}/configuration/jobs ]; then
+    echo "---> Removing sample Jenkins job ..."
+    rm -rf ${JENKINS_DIR}/configuration/jobs
+  fi
+  echo "---> Installing new Jenkins configuration ..."
+  chmod -R go+rw ${JENKINS_INSTALL_DIR}/configuration
```

Comment:
better still use this:
https://github.com/openshift/sti-base/blob/master/bin/fix-permissions

(you'll have to make a copy since this image isn't built on sti-base.  but maybe it should be now that it's an s2i builder?)